src/ops/stream_events.rs

```diff
@@ -0,0 +1,101 @@
+//! Run to output a stream of build events in a machine-parseable form.
+use crate::build_loop::Event;
+use crate::ops::{
+    error::{ok, ExitError, OpResult},
+    get_paths,
+};
+use crate::rpc;
+use crossbeam_channel::{select, unbounded};
+use slog_scope::debug;
+use std::convert::TryInto;
+use std::str::FromStr;
+use std::thread;
+use std::time::Duration;
+
+/// Options for the kinds of events to report
+#[derive(Debug)]
+pub enum EventKind {
+    /// Report only live events - those that happen after invocation
+    Live,
+    /// Report events recorded for projects up until invocation
+    Snapshot,
+    /// Report all events
+    All,
+}
+
+impl FromStr for EventKind {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "all" => Ok(EventKind::All),
+            "live" => Ok(EventKind::Live),
+            "snapshot" => Ok(EventKind::Snapshot),
+            _ => Err(format!("{} not in all,live,snapshot", s)),
+        }
+    }
+}
+
+#[derive(Debug)]
+enum Error {
+    Varlink(rpc::Error),
+    Compat(String),
+}
+
+/// See the documentation for lorri::cli::Command::Shell for more
+/// details.
+pub fn main(kind: EventKind) -> OpResult {
+    // TODO: set up socket path, make it settable by the user
+    debug!("Starting stream_events");
```

Comment:
This seems unnecessary. Nothing has happened yet, so this debug line doesn't really contain any information.